extensions/notebook/src/jupyter/serverInstance.ts

```diff
@@ -143,16 +145,32 @@ export class PerNotebookServerInstance implements IServerInstance {
 		return this._uri;
 	}
 
+	public get attachedEditorCount(): number {
+		return this._attachedEditorCount;
+	}
+
 	public async configure(): Promise<void> {
 		await this.configureJupyter();
 	}
 
+	public incrementAttachedEditorCount(): void {
+		this._attachedEditorCount++;
+		console.log('current attached editor count: ' + this._attachedEditorCount);
+	}
+
 	public async start(): Promise<void> {
 		await this.startInternal();
+		this.incrementAttachedEditorCount();
 	}
 
 	public async stop(): Promise<void> {
 		try {
+			if (this._attachedEditorCount > 1) {
+				this._attachedEditorCount--;
```

Comment:
Shouldn't you always decrement the counter? If it's currently at 1 (so the last editor called stop) then it's going to stay 1 and thus not kill the child process at the end